tests/imbu/integration/imbu_model_lifecycle_tests.py

```diff
@@ -0,0 +1,110 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have purchased from
+# Numenta, Inc. a separate commercial license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import numpy
+import os
+import shutil
+import tempfile
+import unittest
+
+from htmresearch.encoders import EncoderTypes
+from htmresearch.frameworks.nlp.imbu import ImbuModels
+from htmresearch.frameworks.nlp.classification_model import (
+  ClassificationModel
+)
+from htmresearch.frameworks.nlp.classify_network_api import (
+  ClassificationNetworkAPI
+)
+
+
+
+TEST_DATA_DIR = os.path.abspath(
+  os.path.join(
+    os.path.dirname(
+      os.path.realpath(__file__)
+    ),
+    "..", "..", "nlp", "unit", "test_data"
+  )
+)
+
+
+class TestImbuLifecycle(unittest.TestCase):
+
+  def setUp(self):
+    self.dataPath = os.path.join(TEST_DATA_DIR, "sample_reviews_subset.csv")
+
+
+  def _createTempDir(self):
+    tmpDir = tempfile.mkdtemp()
+    self.addCleanup(shutil.rmtree, tmpDir)
```

Comment:
The new cache directory gets created in the call to `imbu.createModel()`. If something in that call breaks the test, the cache directory gets deleted in the test clean up. Instead, if I explicitly delete it later, the cache directory would remain if `imbu.createModel()` breaks the test.